python/pyspark/sql/dataframe.py

```diff
@@ -3008,6 +3008,25 @@ def __getattr__(self, name: str) -> Column:
         jc = self._jdf.apply(name)
         return Column(jc)
 
+    def __dir__(self):
+        """
+        Examples
+        --------
```

Comment:
Should better have some explanation for the example here.

Reply:
added